### ISSUE TITLE
Pointer events: auto in the HTML component makes showing the component on hover tricky

### ISSUE THREAD

**Issue by looeee** [pmndrs/drei#460]
- `three` version: 129
- `@react-three/fiber` version: 7.01
- `@react-three/drei` version: 6.2.0

### Problem description:

I've created a <HTML> component:

```js
export default function Label({visible}) {
  return (
    <Html
      rotation-x={-Math.PI / 2}
      transform
      sprite
      className="pointer-events-none"
    >
      <div
        className={
          visible
            ? "bg-red-400 pointer-events-none"
            : "bg-red-400 pointer-events-none opacity-0"
        }
      >
        <h1 className="pointer-events-none">TESTING</h1>
      </div>
    </Html>
  );
}


```

And I'm showing it on hover like this:

```js
  const [hovered, setHovered] = useState(false);
  const [labelPosition, setLabelPosition] = useState([0, 0, 0]);

  return (
    <group
      {...props}
      dispose={null}
      onPointerOver={(e) => {
        setHovered(true);
        e.stopPropagation();
      }}
      onPointerLeave={(e) => {
        setHovered(false);
        e.stopPropagation();
      }}
    >
      <Label
        visible={hovered ? true : false}
      ></Label>
   </group>)
```

That all works OK, except that the label overlays the mesh and when the mouse goes over the label it's no longer hitting the mesh so hover gets set to false, the label disappears, the raycast hits the mesh, so the label appears again, repeat... in short, flickering occurs. 

...hence all the `pointer-events-none` classes. The problem is, when `transform` is set to true on `<HTML>` the following lines run and set a `pointer-events: auto` on the `transformInnerRef` which I have no control over. 

https://github.com/pmndrs/drei/blob/12e45b7dfffb977a74b90abed92c524ffa2e36f6/src/web/Html.tsx#L190-L204

### Suggested solution:

I'm not sure what the purpose of `pointer-events: auto` on `transformInnerRef` is. Could it be removed?

**Comment by njm222** [comment]
It's needed for pointer events on the HTML, but I think we can remove it since pointer events can still target the children.

So we can:
1. remove pointerEvents and leave it all to the users to decide how to deal with pointer events
2. move pointerEvents to a prop
3. other suggestions..?

I prefer removing it so that it inherits the `pointerEvents: none` from the parent and users can just add pointerEvents to divs that need it.

I can implement it pretty quick if other people are on board?

**Comment by drcmda** [comment]
there is no solution for this, this is a dom/css problem. if you have overlays and they receive events, the underlying layer will not receive anything.

the solution is pretty simple, you need to attach canvas event onto the same parent that contains both the canvas and the overlays.

```
const ref = useRef()
return (
  <div id="full-width-container" ref={ref}>
    <Canvas onCreated={state => {
      state.events.connect(ref.current)
```

now you don't need pointerevents: none, the event will hit both the overlays and the canvas

**Comment by drcmda** [comment]
no idea where it comes from, or why it's there. anyone knows how to properly interpret git blame?

**Comment by njm222** [comment]
Yep found the commit:  https://github.com/pmndrs/drei/commit/be9ae6966b54ccac4f6c0b3a0f29d32e61ed79c7#diff-eb7dcf4ddfc3cb0afb087ecbfb9543791f4153b544af8c5081da23c29e4a2a6b

I think it is there under the assumption that we need it and it makes sense removing it so that the user is able to decide if they want it or not

**Comment by looeee** [comment]
> 1. remove pointerEvents and leave it all to the users to decide how to deal with pointer events

I think this is the best solution unless there is another reason for it being there.

**Comment by drcmda** [comment]
i dont understand this

> While on desktop the events are forwarded easily between html and webgl, on mobile you have to perform a double tap in order to switch between them (what I mean with "forwarding events" is simply setting pointerEvents to "none" or "auto" on the webgl container)

if we remove it, are users still allowed to fix this use case. and if it's mobile and it really doesnt work without that prop, doesn't it mean it must be in there?

**Comment by joshuaellis** [comment]
I believe this was resolved in #492
